### PR TITLE
feature/change_color_of_footer

### LINKED ISSUE #2435: Modify background color of footer
Following Footer elementes should be changed:

axa-footer (PL)
axa-footer-small-aem (PL)

The new colors should be:

Ocean Blue: #4976BA
Ocean Blue Dark: #3A5E95

### PR DESCRIPTION
**Pull request by marcwelz**
Fixes #2435

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch-webhub-cloud/pattern-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: src/components/30-organisms/footer/index.scss [hunk 1]
```diff
@@ -19,19 +19,23 @@
   }
 
   &__content {
+    display: flex;
+    flex-direction: column-reverse;
+
     @media (min-width: $breakpoint-xs) {
       display: grid;
       grid-template-areas:
-        'links'
```
Comment (thomas-krueger-axa) [inline src/components/30-organisms/footer/index.scss]: Why does it need this change? It's not part of the story.

File: src/components/30-organisms/footer/index.scss [hunk 2]
```diff
@@ -52,8 +56,11 @@
   }
 
   &__social-media {
+    border-bottom: 1px solid rgba($color-prim-white, 0.25);
```
Comment (thomas-krueger-axa) [inline src/components/30-organisms/footer/index.scss]: Why does it need this change? It's not part of the story.

File: src/components/30-organisms/footer/index.scss [hunk 3]
```diff
@@ -111,20 +118,20 @@
...
+    background-color: $color-ocean-blue;
+    padding: 20px 30px;
     width: 100%;
     text-align: left;
 
     border: none;
     outline: 0;
 
-    border-bottom: 1px solid $color-azure;
+    border-bottom: 1px solid rgba($color-prim-white, 0.25);
```
Comment (thomas-krueger-axa) [inline src/components/30-organisms/footer/index.scss]: Why does it need this change? It's not part of the story.

File: src/components/30-organisms/footer/index.scss [hunk 4]
```diff
@@ -111,20 +118,20 @@
...
     text-transform: uppercase;
 
     svg {
-      width: 10px;
```
Comment (thomas-krueger-axa) [inline src/components/30-organisms/footer/index.scss:126]: Why does it need this change? It's not part of the story.

File: src/components/30-organisms/footer/index.scss [hunk 5]
```diff
@@ -170,13 +177,13 @@
...
 
       overflow: hidden;
-      background-color: $color-azure;
+      background-color: $color-ocean-blue-dark;
 
       &-list-item {
         @include text(t4);
 
         padding: 0;
-        border-bottom: 1px solid $color-dark-indigo;
```
Comment (thomas-krueger-axa) [inline src/components/30-organisms/footer/index.scss:179]: Why does it need this change? It's not part of the story.

File: src/components/30-organisms/footer/index.scss [hunk 6]
```diff
@@ -190,6 +197,7 @@
           width: 165px;
           margin-top: 8px;
           padding: 5px 0;
+          border: none;
```
Comment (thomas-krueger-axa) [inline src/components/30-organisms/footer/index.scss]: Why does it need this change? It's not part of the story.

**Review by thomas-krueger-axa — CHANGES_REQUESTED** [review]
Hi @marcwelz , thank you for the changes.

First of all, all changes in the pattern lib need an issue in Github. So please add an issues and link it to this PR.

Please choose a concise statement for your commit messages that will help others understand what has been changed.

On my local environment the tests run on errors.

The CI pipeline has an error, maybe the same as on my local machine.

File: src/components/30-organisms/footer/index.scss (hunk 4 above)
Comment (marcwelz) [reply src/components/30-organisms/footer/index.scss:126]: I was testing something and accidentally pushed it, I will remove it

**Review by thomas-krueger-axa — DISMISSED** [review]

File: src/components/30-organisms/footer/e2e.js [hunk 7]
```diff
@@ -20,4 +20,163 @@ test.describe('footer', () => {
...
+    expect(
+      await instagramSocialSvg.evaluate(
+        el => window.getComputedStyle(el).height
+      )
+    ).toBe('25px');
+    expect(await instagramSocialSvg.getAttribute('viewBox')).toBe('0 0 25 25');
+    expect(await instagramSocialSvg.count()).toBe(1);
+  });
+
+  test('should render accordion only in mobile mode', async ({ page }) => {
```
Comment (markus-walther) [inline src/components/30-organisms/footer/e2e.js:75]: This test would profit from comments. Set viewport 1px above maximum mobile width breakpoint, measure that no accordion is displayed, set viewport exactly to breakpoint width, measure that accordion is visible. Four different steps, 4 comments. Makes the same test much easier to read and review. Apply in the same spirit to other cases that are not absolutely trivial.

File: src/components/30-organisms/footer/e2e.js [hunk 8]
```diff
@@ -20,4 +20,163 @@ test.describe('footer', () => {
...
+    await page.goto(fixtureURL('components-footer--footer'));
+    page.setViewportSize({ width: 576, height: 400 });
+
+    const accordion = page.locator('.o-footer__accordion-button-caret').nth(0);
```
Comment (markus-walther) [inline src/components/30-organisms/footer/e2e.js]: What does the nth(0) do? I can guess and look it up, or you write a small comment, e.g. // nth(0): pick the first accordion item.

**Review by thomas-krueger-axa — DISMISSED** [review]
Hi @marcwelz , thanks for your changes.

While running the tests i get following error:
>1 failed
    [chromium] › src/components/30-organisms/footer/e2e.js:85:3 › footer › should correctly open accordion on click

Also it seems you migrated some tests, but didn't delete the old duplicates.

And maybe you can check or fix this warning:
>Consider splitting slow test files to speed up parallel execution

File: src/components/30-organisms/footer/index.js [hunk 9]
```diff
@@ -20,6 +20,37 @@ const _setMaxHeightToZero = panel => {
   panel.style.maxHeight = '0px';
 };
 
+// hence there is no re-render to set the caret-open-class
```
Comment (markus-walther) [inline src/components/30-organisms/footer/index.js]: This comment can not be understood properly. Rephrase: Caret state change *bypasses* render() and directly manipulates dom-element classes on purpose here, because the height measurement of open accordion items conflicted with a model-based always-rerender approach in the past. TODO: Explore how to get rid of height measurement or make it compatible with always-rerender, then refactor and remove this code in the future.

File: src/components/30-organisms/footer/index.js [hunk 10]
```diff
@@ -20,6 +20,37 @@ const _setMaxHeightToZero = panel => {
...
+// it has to be done by manually adding the class to the
+// element via javascript
+const _changeStateOfCaret = (caret, isActive) => {
```
Comment (markus-walther) [inline src/components/30-organisms/footer/index.js:32]: Please explain once, what a caret is: The caret is a little UI icon visually indicating accordion open/closed state.

File: src/components/30-organisms/footer/index.js [hunk 11]
```diff
@@ -331,25 +357,28 @@ class AXAFooter extends InlineStyles {
...
       } = panel;
+
       // set maxHeight to exactly the height of all elements combined
       panel.style.maxHeight = `${Math.ceil(offsetHeight * children.length)}px`;
+
+      const caret = ev.currentTarget.parentNode.querySelector(
+        '.js-footer__accordion-button-caret'
+      );
+
+      // sets the class for the caret
```
Comment (markus-walther) [inline src/components/30-organisms/footer/index.js]: // sets the class for the caret **open state**

**Review by markus-walther — CHANGES_REQUESTED** [review]
Hi Marc,

a few findings regarding comments.

Very close to done, good work!

--Markus

File: src/components/30-organisms/footer/e2e.js [hunk 12]
```diff
@@ -20,4 +22,172 @@ test.describe('footer', () => {
...
+  test('should render accordion only in mobile mode', async ({ page }) => {
+    await page.goto(fixtureURL('components-footer--footer'));
+    // Set viewport 1px above maximum mobile width breakpoint
+    page.setViewportSize({ width: 576, height: 400 });
+
+    const accordionFirstButton = page
+      .locator('.o-footer__accordion-button-caret')
+      .nth(0);
+
+    // Measure that accordeon is not visible on desktop breakpoint
```
Comment (markus-walther) [inline src/components/30-organisms/footer/e2e.js:84]: Good!

File: src/components/30-organisms/footer/index.js [hunk 13]
```diff
@@ -20,6 +20,43 @@ const _setMaxHeightToZero = panel => {
   panel.style.maxHeight = '0px';
 };
 
+/**
+ * The caret is a little UI icon visually indicating accordion open/closed state.
```
Comment (markus-walther) [inline src/components/30-organisms/footer/index.js:24]: Good!

**Review by markus-walther — APPROVED** [review]
Marc,

good work!

Cheers, Markus
